src/containers/leaderboard/Leaderboard.js

```diff
@@ -112,6 +113,7 @@ class Leaderboard extends React.Component {
     results = results.filter(profile => {
```

Comment:
Such code could be rewritten in cascade style:
```javascript
results = results.filter(
    profile => profile.name.toLowerCase().indexOf(this.state.nameFilter.toLowerCase()) >= 0
);
```

But it could be improved with method `includes`, precalculated name filter value and destructuring.
```javascript
const nameFilter = this.state.nameFilter.toLowerCase();
results = results.filter(({name}) => name.toLowerCase().includes(nameFilter));
```